sklearn/utils/validation.py

```diff
@@ -506,14 +507,38 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    # Indices Datatype regulation
+    _check_large_sparse(array, accept_large_sparse)
+
     return array
 
 
+def _check_large_sparse(array, accept_large_sparse=False):
+    """
+    Indices Regulation for CSR Matrices.
+    Only Int32 are supported for now
+
+    """
+    if accept_large_sparse is False and type(array) == sp.csr.csr_matrix:
```

Comment:
we need to be checking CSC matrices and COO matrices also, at least. Why not just use isspmatrix

Reply:
will do this in next commit

Reply:
@jnothman i checked for almost all the algorithms with CSC and COO matrices, it passed all the test cases.

I analyzed why these matrices are behaving like these and found that during preprocessing CSC and COO matrices are converted to CSR matrices using _ensure_sparse_format function

as _check_large_sparse is called after _ensure_sparse_format, therefore I don't feel the need to handle CSC and COO matrices cases.

I will do extensive check in order to find out whether any algorithm is prioritizing COO/CSC over CSR, if you know any plz share (it would ease the implementation)

Reply:
I think it would be a good idea to just handle other cases here, as `_check_large_sparse` can and is being used as a standalone validation routine.

Please avoid `is False`. Just use `if not accept_large_sparse `